src/plugins/janus_sip.c

```diff
@@ -7392,6 +7419,56 @@ gpointer janus_sip_sofia_thread(gpointer user_data) {
 	return NULL;
 }
 
+/* check peer rtp has rfc2833 and push event */
+void janus_sip_check_rfc2833(janus_sip_session *session, char *buffer) {
+	janus_rtp_header *rtp_header = (janus_rtp_header *)buffer;
+	if (rtp_header->type != JANUS_RTP_RFC2833_DEFAULT_PAYLOAD_TYPE)
+		return;
+	char *payload_buffer = buffer + JANUS_RTP_HEADER_LENGTH + rtp_header->csrccount * sizeof(uint32_t);
+	janus_rtp_rfc2833_payload *rfc2833_payload = (janus_rtp_rfc2833_payload *)payload_buffer;
```

Comment:
You're not checking if the payload is large enough to allow for the casting. If it isn't, it will crash Janus.